include/circt/Dialect/RTL/Structure.td

```diff
@@ -0,0 +1,88 @@
+def RTLModuleOp : RTLOp<"module",
+      [IsolatedFromAbove, FunctionLike, Symbol, RegionKindInterface,
+       SingleBlockImplicitTerminator<"DoneOp">]> {
+  let summary = "RTL Module";
+  let description = [{
+    The "rtl.module" operation represents a Verilog module, including a given
+    name, a list of ports, and a body that represents the connections within
+    the module.
+  }];
+  let arguments = (ins);
+  let results = (outs);
+  let regions = (region SizedRegion<1>:$body);
+
+  let skipDefaultBuilders = 1;
+  let builders = [
+    OpBuilder<"OpBuilder &builder, OperationState &result, StringAttr name, "
+              "ArrayRef<std::tuple<StringAttr, Type, StringAttr>> ports">
+  ];
```

Comment:
This is good as a first step, but I'd prefer for `rtl.module` to represent outputs as operands to the "done" op (which can then be renamed to return or something) instead of being BB operands that get connected to.  Landing this patch first and then making the second step makes a lot of sense given the size of this patch.

The rationale for the FIRRTL module/instance stuff is set up the way it is that it made sense to follow the internal design of FIRRTL consistently there.  We can then lower from FIRRTL -> RTL module/instance to clean this up and make it explicit.

For completeness, I'll mention that inout arguments should be modeled as bb arguments like inputs, and should have connects to them.